source/test/TestApp/TestApp.cpp

```diff
@@ -89,17 +103,21 @@ namespace PlayFabUnit
             std::bind(&TestApp::OnPostReportError, this, std::placeholders::_1, std::placeholders::_2),
             &testRunner.suiteTestReport);
 
-        while (cloudResponse.empty())
+        for (int i = 0; i < CLOUDSCRIPT_TIMEOUT_MS; i+= CLOUDSCRIPT_TIMEOUT_INCREMENT)
         {
-            std::this_thread::sleep_for(TimeValueMs(100));
+            std::this_thread::sleep_for(std::chrono::milliseconds(CLOUDSCRIPT_TIMEOUT_INCREMENT));
+            if (!cloudResponse.empty())
```

Comment:
sleep_for in a loop + cloudResponse.empty() check looks like an overly verbose way to implement a condition_variable (I'd say your current code is anti pattern that enables concurrency bugs)

recommend looking at std::condition_variable specifically std::condition_variable::std::wait_for

it will let you wait for a resource to become available (cloudResponse) and will avoid the need to declare the CLOUDSCRIPT_TIMEOUT_INCREMENT  variable

Reply:
I have a bug about an occasional TestApp deadlock issue, and your suggestion resonates with that bug.
I'm going to copy your comments into that bug.